apps/shared/avifjpeg.c

```diff
@@ -599,66 +597,51 @@ static avifBool avifJPEGParseGainMapXMPProperties(const xmlNode * rootNode, avif
 
     avifGainMapMetadataDouble metadataDouble;
     // Set default values from Adobe's spec.
-    metadataDouble.baseRenditionIsHDR = AVIF_FALSE;
-    metadataDouble.hdrCapacityMin = 1.0; // exp2(0) = 1
-    metadataDouble.hdrCapacityMax = exp2(1);
+    metadataDouble.backwardDirection = AVIF_FALSE;
+    metadataDouble.baseHdrHeadroom = 0.0;
+    metadataDouble.alternateHdrHeadroom = 1.0;
     for (int i = 0; i < 3; ++i) {
-        metadataDouble.gainMapMin[i] = 1.0; // exp2(0) = 1
-        metadataDouble.gainMapMax[i] = exp2(1.0);
-        metadataDouble.offsetSdr[i] = 1.0 / 64.0;
-        metadataDouble.offsetHdr[i] = 1.0 / 64.0;
+        metadataDouble.gainMapMin[i] = 0.0;
+        metadataDouble.gainMapMax[i] = 1.0;
+        metadataDouble.baseOffset[i] = 1.0 / 64.0;
+        metadataDouble.alternateOffset[i] = 1.0 / 64.0;
         metadataDouble.gainMapGamma[i] = 1.0;
     }
 
+    AVIF_CHECK(avifJPEGFindGainMapPropertyDoubles(descNode, "HDRCapacityMin", &metadataDouble.baseHdrHeadroom, /*numDoubles=*/1));
+    AVIF_CHECK(avifJPEGFindGainMapPropertyDoubles(descNode, "HDRCapacityMax", &metadataDouble.alternateHdrHeadroom, /*numDoubles=*/1));
+    AVIF_CHECK(avifJPEGFindGainMapPropertyDoubles(descNode, "OffsetSDR", metadataDouble.baseOffset, /*numDoubles=*/3));
+    AVIF_CHECK(avifJPEGFindGainMapPropertyDoubles(descNode, "OffsetHDR", metadataDouble.alternateOffset, /*numDoubles=*/3));
+    AVIF_CHECK(avifJPEGFindGainMapPropertyDoubles(descNode, "GainMapMin", metadataDouble.gainMapMin, /*numDoubles=*/3));
+    AVIF_CHECK(avifJPEGFindGainMapPropertyDoubles(descNode, "GainMapMax", metadataDouble.gainMapMax, /*numDoubles=*/3));
+    AVIF_CHECK(avifJPEGFindGainMapPropertyDoubles(descNode, "Gamma", metadataDouble.gainMapGamma, /*numDoubles=*/3));
+
+    // Should have HDRCapacityMax > HDRCapacityMin.
+    if (metadataDouble.alternateHdrHeadroom <= (double)metadataDouble.baseHdrHeadroom) {
```

Comment:
Already a `double`, no need for a cast.

Consider `AVIF_CHECK` to keep the same comparison symbol `>` as the comment.